### DISCUSSION TITLE
Child Window Position ignored on Ubuntu 22.04/Linux Lite 5.8

### DISCUSSION THREAD

**Discussion by rexn8r** [AvaloniaUI/Avalonia#8237] [Q&A]
Hello

i am experimenting with Avalonia UI in Visual Studio 2022. I have created a sample application with Libvlcsharp avalonia nuget and below is the link of git repo;

https://github.com/rexn8r/AvaloniaTest.git

Its kind of kiosk application without titlebar and windows menu buttons. I have 3 avalonia windows. 

ParentWindow (no decoration, manual start up location, full screen)
MainWindow (hosting libvlcsharp videoview with no decoration, manual startup location and launched by Parentwindow at 0,0 position and custom size and made child of Parent window)
ImageWindow (hosting AvaloniaGif control with no decoration, manual startup location and launched by Parentwindow at 0,0 position and custom size and made child of Parent window)

When i execute the program on windows, it works OK and the child windows are rendered at 0,0 positions with custom size.

I have created a linux package and when i run the app on Ubuntu 22.04 as well as Linux Lite 5.8, the child window positions are offset from the top/left corner.

I have attached a screenshot of the app (taken on Linux Lite).

![ss](https://user-images.githubusercontent.com/12812556/171333418-0b933545-ca11-48ae-9e82-996dcdc969a2.png)

can someone please guide me how to resolve this issue? 

Thanks & Regards
Rex

**Comment by rexn8r** [comment]
Hello Again,

Interesting find.

On Linux Lite 5.8, I have installed VS Code with .net core and have downloaded following avalonia/libvlcsharp sample from following repo;

https://github.com/donandren/vlcsharpavalonia

I have then modified the MainWindow.xaml and added code in the initialisation to set window position to 0,0 as follows;

<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:v="clr-namespace:LibVLCSharp.Avalonia;assembly=LibVLCSharp.Avalonia"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="LibVLCSharp.Avalonia.Sample.MainWindow"
        HasSystemDecorations="False"
		CanResize="False"
		SystemDecorations="None"
		WindowStartupLocation="Manual"
		WindowState="Normal"
        Title="LibVLCSharp.Avalonia.Sample" Height="500" Width="800">
  <StackPanel HorizontalAlignment="Center" VerticalAlignment="Center">
    <Button x:Name="example1" Content="Example 1"  Width="100"/>
    <Button x:Name="example2" Content="Example 2"  Width="100"/>
  </StackPanel>
</Window>

cs code;

public MainWindow()
        {
            InitializeComponent();

            this.Get<Button>("example1").Click += (s, e) =>
            {
                var w = new Example1();
                w.Show();
            };

            this.Get<Button>("example2").Click += (s, e) =>
            {
                var w = new Example2();
                w.Show();
            };

            **this.Position = PixelPoint.FromPoint(new Point(0, 0), 0);**    **--->> added this line**

        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
        }

When i build the app and run it, i can see following behaviour;

1. If the Terminal window is open at 0,0 position then the sample app window opens with offset as shown in following screenshot;

![image](https://user-images.githubusercontent.com/12812556/171341984-053f1f2a-c1ed-411d-9bf9-7c573e3777fc.png)

2. Now if i move the Terminal window position to lets say extreme right of the desktop and then run the app, the app window opens at 0,0 position as shown in the following screenshot;

![image](https://user-images.githubusercontent.com/12812556/171342267-f150ce0d-fa38-4d9f-8c67-ee9c5b511827.png)

So its like if there are existing windows already open on the desktop at different position with in the avalonia application client area, the application opens at the position with sum of offset of all the existing opened windows. so looks like the linux windows desktop manager behaves differently then windows os desktop manager.

can any one throw a light as to how to force avalonia application to always open @ 0,0 position and also how to resolve child window position problem as mentioned above?

thanks
rex

**Reply by kekekeks** [reply]
Most of window managers completely ignore the initial window position supplied by the application and reposition the window as they see fit. You can try handing `PoisitonChanged` event and moving the window again. Window managers _usually_ comply to reposition requests when the window is already shown.

**Reply by NickBarton** [reply]
> Most of window managers completely ignore the initial window position supplied by the application and reposition the window as they see fit. You can try handing `PoisitonChanged` event and moving the window again. Window managers _usually_ comply to reposition requests when the window is already shown.

Brilliant, was struggling with this on Linux, works a treat.

**Comment by rexn8r** [comment]
hi @kekekeks 

that did the trick. thanks for that.

regards
rex
